### PR TITLE
Adding support for glint-language-server for js files

### PR DESCRIPTION
**Pull request by erinsinger93**
I'm sure there will be plenty of things to fix-up here, but I wanted to get a PR going to start collecting feedback. I have it up and running locally, and it's looking good.

Summary:
- Glint config now includes js files by default (+added test coverage)
- Transform manager checks for files ending in ts or js
- Language server index: 
  - Checks for jsconfig.json or tsconfig.json for configPath
  - Filters root filenames by ts or js
- Language server util parseConfigFile now works for both jsconfig or tsconfig
- Ember loose environment permits ts or js files for possible script paths (+ added test coverage)
  - I moved around a lot of files since previously the environment didn't have any associated jest tests 
-  Template-to-typescript: 
  - When useJsDoc is true, emitTemplateBoilerplate should emit jsdoc rather than typescript (+added test coverage) 
- Updated vscode package to support js files
- Added jsdocs in the dummy glimmer-js app
- Added js-glimmerx-app smoketests to packages/vscode

Followup:
- I would like to rename ts specific files/functions/variables to be script-type agnostic

### CODE REVIEW

File: packages/core/src/language-server/util/index.ts
Comment (jamescdavis) [inline packages/core/src/language-server/util/index.ts:28]:
```suggestion
    ? configPath.replace(new RegExp(`${jsConfig ? 'jsconfig' : 'tsconfig'}\\.json$`), '')
```

File: packages/core/src/language-server/util/index.ts
Comment (jamescdavis) [inline packages/core/src/language-server/util/index.ts:30]:
```suggestion
    ? configPath.replace(new RegExp(`${jsConfig ? 'jsconfig' : 'tsconfig'}\\.json$`), '')
```

File: packages/transform/src/template-to-typescript.ts
Comment (dfreeman) [inline packages/transform/src/template-to-typescript.ts:91]: I'm inclined to say it's probably ok to just not support type params for JS sources. The alternative is getting into parsing the JSDoc for the class ourselves and sorting through `@template` pragmas and their friends, which is possible but definitely a nontrivial amount of work.

File: packages/transform/src/template-to-typescript.ts
Comment (dfreeman) [inline packages/transform/src/template-to-typescript.ts:88]: [nit] Could this be broken into two `emit.text` calls? This took me a couple of tries to read 😄 

```suggestion
        if (contextType) {
          emit.text('/** @type {unknown} */ (');
        }
        emit.text(`(/** @type {typeof import("${typesPath}")} */ ({})).template(function(`);
```

File: packages/vscode/__fixtures__/js-glimmerx-app/src/SimpleComponent.js
Comment (dfreeman) [inline packages/vscode/__fixtures__/js-glimmerx-app/src/SimpleComponent.js:10]: Generally the terminology we've been encouraging is `Signature` to encompass the thing that contains `Args`/`Yields`/etc.

```suggestion
/**
 * @typedef SimpleComponentSignature
 * @property {object} Args
 * @property {string} Args.message
 */

/** @extends {Component<SimpleComponentSignature>} */
```

**Review by dfreeman — COMMENTED** [review]
This is _super_ cool. I pulled down your branch and started playing with it, and there was an embarrassingly goofy grin on my face when I wrote this just worked in a plain `.js` file with no extra annotations at all 😄 

image

The overall content of this change makes total sense to me; the comments I have are mostly in-the-weeds nitpicks. Thank you for all your hard work on this! I'm super excited to see it land.

File: packages/core/src/language-server/index.ts
Comment (erinsinger93) [reply packages/core/src/language-server/index.ts:20]: Sounds good. Done

File: packages/core/src/language-server/index.ts
Comment (erinsinger93) [reply packages/core/src/language-server/index.ts:27]: Do we want to make a separate isScript helper for the language-server folder? Or is it ok if I pull in a helper from the transform package?

Comment (erinsinger93) [reply packages/core/src/language-server/index.ts:27]: Actually I see we're pulling in other helpers from that package in other files, so I'll follow suit here

File: packages/environment-ember-loose/-private/environment/index.ts
Comment (erinsinger93) [reply packages/environment-ember-loose/-private/environment/index.ts:60]: Agreed -- I refactored this a few times and somehow ended up with silly logic :P I'll update to your suggestion

File: packages/transform/src/index.ts
Comment (erinsinger93) [reply packages/transform/src/index.ts:105]: When I try using the language server on the js-glimmerx app without this line I get a lot of unexpected errors and warnings, for example:
```<component-name> is declared but its value is never read.``` 
or
```
Type 'GlimmerComponent<{}>' is not assignable to type 'TemplateComponent<ComponentAsArgArgs>'.
  Type 'GlimmerComponent<{}>' provides no match for the signature 'new (): { [Invoke]: (args: { title: string; }) => AcceptsBlocks<EmptyObject, null>; [Context]: TemplateContext<null, { title: string; }, EmptyObject, null>; }'.
```

File: packages/transform/src/template-to-typescript.ts
Comment (erinsinger93) [reply packages/transform/src/template-to-typescript.ts:91]: Sounds good, thanks for the feedback! I'll remove the investigation comment then

File: packages/transform/src/index.ts
Comment (dfreeman) [reply packages/transform/src/index.ts:105]: Weeeeeird. Ok! I may pull this thread sometime in the future to see if I can understand what's happening there, but keeping this around for now seems fine 👍

**Review by dfreeman — APPROVED** [review]
Fantastic! 🎉 

We really need to get moved over to Gitbook or some other structured setup for docs, but for now do you think it makes sense (either here or in a follow-on PR) to update the README with instructions for use with a JS project?

I've been working in parallel on template-only component support for Ember projects and have been holding off on finishing it up to avoid conflicts with this PR. Once this is merged I'll aim to get that wrapped up this week so we can get v0.6.0 released!

**Comment by erinsinger93** [comment]
I'll author the README in a followup change. I've made an issue to track the work

**Pull request merged**
